src/EditorFeatures/Core.Wpf/InlineRename/UI/InlineRenameAdornmentProvider.cs

```diff
@@ -76,7 +76,7 @@ internal class InlineRenameAdornmentProvider : IWpfTextViewConnectionListener
             _listenerProvider = listenerProvider;
             if (smartRenameSessionFactory is not null)
             {
-                _smartRenameSessionFactory = new Lazy<ISmartRenameSessionFactoryWrapper>(() => ISmartRenameSessionFactoryWrapper.FromInstance(smartRenameSessionFactory));
+                _smartRenameSessionFactory = new Lazy<ISmartRenameSessionFactoryWrapper>(() => ISmartRenameSessionFactoryWrapper.FromInstance(smartRenameSessionFactory.Value));
```

Comment:
Need to call `Value` here otherwise it going to convert 'Lazy' to object, but since it's a nested lazy it's still going be lazy